File: frontend/app/src/main/scala/app/domain/followers/RelationShipLink.scala
package app.domain.followers

import app.domain.core.AbstractLink
import jfx.core.meta.Meta
import jfx.json.JsonType

import scala.scalajs.js

@JsonType("followers-list")
class RelationShipLink(
  var rel: String = "",
  var url: String = "",
  var method: String = "GET"
) extends AbstractLink {

  override def name: String = "Followers"

  override def icon: String = "1k_plus"
}

File: frontend/app/src/main/kotlin/app/domain/followers/RelationShipLink.kt
package app.domain.followers

import app.domain.core.AbstractLink
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable


@Serializable
@SerialName("followers-list")
class RelationShipLink(
    override val id: String,
    override val rel: String,
    override val url: String,
    override val method: String = "GET"
) : AbstractLink() {

    override val name: String = "Followers"
    override val icon: String = "1k_plus"

}